### ISSUE TITLE
Clicking older or newer posts button not show list of blog post but show all categories instead

### ISSUE THREAD

**Issue by aaronamm** [OrchardCMS/OrchardCore#10384]
### Describe the bug
- On blog posts in a category page, if there are pagination buttons, clicking older or newer posts button bring users to all categories page which is not expected behavior.
- Given we have 6 blog posts and set page size to 2.
- We have 3 pages (2 post items per page)
- We can't get 3rd page by clicking **Older Posts** button but we get all categories page instead.
- I am not sure if it happens when we are about to go to first or last page only.

### To Reproduce
- Setup or new Orchard Core CMS website with a blog recipe or use my existing demo website:
  - https://orchard-core-demo.azurewebsites.net/
  - username: admin
  - password: 12345Abc$
  - source code https://github.com/aaronamm/OrchardCoreDemo
  - Orchard version 1.0
  - database SQLite
- Login in as admin.
- Go to blog menu.
- Clone an existing blog post to get 6 post items in total.
- Publish all content items.
- Go to Configuration > Settings > General
- Change page size to **2** to get **3 pages** on frontend.
- Go back to home page and browse all blog posts in travel category (e.g. https://orchard-core-demo.azurewebsites.net/categories/travel)
- You will find a simple pagination with **Older Posts** button.
- Click **Older Posts** button to go to 2nd page.
- Click **Older Post** button again to go to 3rd page.
- **You will be redirect to all categories page that list all categories (e.g. https://orchard-core-demo.azurewebsites.net/categories) which is not what we want.**
- On a page that has a Newer Posts button, try to click it and you will get the same behavior.

### Expected behavior
- Show 3rd page and 1st page instead of showing all categories page
- Do not show "Older Posts" or "Newer Posts" button if there is no blog post, I think it is existing  behavior.

### Screenshots

blog posts in a category page

1st page
![image](https://user-images.githubusercontent.com/2938310/135571627-7dd039bc-ac7d-401d-a326-27a78090e92b.png)

2nd page
![image](https://user-images.githubusercontent.com/2938310/135571756-347710dd-8238-454e-ad8e-53f10b243bd1.png)

Not go to 3rd page but get all categories page instead

![image](https://user-images.githubusercontent.com/2938310/135571836-e80234ce-3426-4e95-b363-985bbea6bf7e.png)

**Comment by Skrypt** [comment]
There is something wrong with the url of the pager. I can see that this part of the url gets removed on the last page "?jsonPath=TaxonomyPart.Terms%5B0%5D&".

**Comment by deanmarcussen** [comment]
Yes, I think there is a bug with url generation when using the pager with Contained Items

**Comment by dohoangtung** [comment]
The bug still remain in Orchard v1.3.0. Any workaround for this ?

**Comment by Skrypt** [comment]
This issue needs to be fixed for 1.4 if anyone has time to work on a PR.

**Comment by aaronamm** [comment]
@Skrypt thanks for your contribution.
